tools/pd-ctl/pdctl/command/global.go

```diff
@@ -33,43 +35,91 @@ import (
 const (
 	pdControlCallerID = "pd-ctl"
 	pingPrefix        = "pd/api/v1/ping"
+	clusterPrefix     = "pd/api/v1/cluster"
 )
 
+func initTLSConfig(caPath, certPath, keyPath string) (*tls.Config, error) {
+	tlsInfo := transport.TLSInfo{
+		CertFile:      certPath,
+		KeyFile:       keyPath,
+		TrustedCAFile: caPath,
+	}
+	tlsConfig, err := tlsInfo.ClientConfig()
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	return tlsConfig, nil
+}
+
 // PDCli is a pd HTTP client
 var PDCli pd.Client
 
-// SetNewPDClient creates a PD HTTP client with the given PD addresses and options.
-func SetNewPDClient(addrs []string, opts ...pd.ClientOption) {
+// shouldInitPDClient checks whether we should create a new PD client according to the cluster information.
+func shouldInitPDClient(cmd *cobra.Command) (bool, error) {
+	if PDCli == nil {
+		return true, nil
+	}
+	// Use PD client to get the current cluster information.
+	currentClusterInfo, err := PDCli.GetCluster(cmd.Context())
+	if err != nil {
+		return false, err
+	}
+	// Use HTTP request to get the new cluster information.
+	newClusterInfoJSON, err := doRequest(cmd, clusterPrefix, http.MethodGet, http.Header{})
+	if err != nil {
+		return false, err
+	}
+	newClusterInfo := &metapb.Cluster{}
+	err = json.Unmarshal([]byte(newClusterInfoJSON), newClusterInfo)
+	if err != nil {
+		return false, err
+	}
+	return currentClusterInfo.GetId() == 0 || newClusterInfo.GetId() != currentClusterInfo.GetId(), nil
+}
+
+// InitNewPDClient creates a PD HTTP client with the given PD addresses.
+func InitNewPDClient(cmd *cobra.Command) error {
+	if should, err := shouldInitPDClient(cmd); !should || err != nil {
+		return err
+	}
 	if PDCli != nil {
 		PDCli.Close()
 	}
-	PDCli = pd.NewClient(pdControlCallerID, addrs, opts...)
+	PDCli = pd.NewClient(pdControlCallerID, getEndpoints(cmd))
+	return nil
 }
 
-// TODO: replace dialClient with PDCli
+// InitNewPDClientWithTLS creates a PD HTTP client with the given PD addresses and TLS config.
+func InitNewPDClientWithTLS(cmd *cobra.Command, caPath, certPath, keyPath string) error {
+	if should, err := shouldInitPDClient(cmd); !should || err != nil {
+		return err
+	}
```

Comment:
we need to consider when old cluster is disconnected, maybe need to renew it directly